sql/catalyst/src/main/scala/org/apache/spark/sql/catalyst/util/DateTimeUtils.scala

```diff
@@ -350,22 +356,58 @@ object DateTimeUtils {
       segments(6) /= 10
       digitsMilli -= 1
     }
+    // This step also validates time zone part
+    val zoneId = tz match {
+      case None => timeZoneId
```

Comment:
We can return None here, then we don't need the parameter `timeZoneId: Option[ZoneId]`

Reply:
in `stringToTimestamp` we can do `zoneid.getOrElse(timeZoneId)`

Reply:
Done, thanks.